### ISSUE TITLE
[Feature request] Insert new chunk  button for qmd files

### ISSUE THREAD

**Issue by danieltomasz** [quarto-dev/quarto-vscode#115]
RStudio has an option to insert new code chunk from the upper bar, jupyter notebook also have it.
Would be great to have this option in similar  option in VSC when editing qmd files.

I see two potential solutions:

1) Additional button above the existing single chunk menu (run cell, run next cell), in a similar  way this is solved in the jupyter notebook
2) Add a button to add a new chunk along the render button, then the chunk will be inserted in  a place of cursor . Default language might be inferred on the basis of the current file, there might also more options along the default one like in the `Render` menu

**Comment by jjallaire** [comment]
FWIW we do have a command/shortcut for this it's **Quarto: Insert Code Cell** and you can execute it with Cmd+Shift+I

Agree though it would be good to have a button for this too!

**Comment by danieltomasz** [comment]
Knowing about the command  is really useful (although I needed to change the shortcut because some other app/plugin stole it :). 
From my personal experience, buttons for the people with less coding background, especially those just starting with the reproducible workflow in academia might be really useful :) 

Thank you for your  hard work! :)

**Comment by jjallaire** [comment]
Yes, agree 100%.

**Comment by mcanouil** [comment]
> buttons for the people with less coding background, especially those just starting with the reproducible workflow in academia might be really useful 

I am not sure, they would have chosen VSCode over RStudio as their starting IDE 😅
I personally do not think adding a new button is a great idea for VSCode, which is more focussed on shortcuts, snippets and the command palette.
For example, why not also add fenced divs buttons? 🤔
I think RStudio is more suitable to get those additional buttons as part of the visual editor toolbar.

**Comment by danieltomasz** [comment]
People choose VSCode to work with python, for example neuroscience is Matlab/Python, R is more cognitive psychology;

Still people like jupyter notebooks, and in principle qmd notebooks should provide similar functionality ...

**Comment by mcanouil** [comment]
I see (I am not very familiar with Jupyter Notebooks).
I would rather have it as an opt-in / opt-out option then.
(The render button is already in a place where you can easily press the button by mistake when many opened files. Also the case from VSCode-r extension and RMarkdown files.)

**Comment by danieltomasz** [comment]
Yes, I am completely for the option opt-in/opt-out option!

**Comment by jjallaire** [comment]
I agree that the button is strictly better for scientists as opposed to more hardcore programmer / text-editor issues. That said, the ethos of VS Code is to be pretty sparing with buttons and rely more on commands and keybindings (the Jupyter notebook is a special case UI -- you can't add that sort of toolbar to a normal text file editor).

**Comment by mcanouil** [comment]
Additional thoughts on this:

Contextual menu in VSCode seems to be customisable since it reacts/changes depending on contents, maybe adding a new menu in contextual menu for Quarto documents could be a better solution.
With that solution, other options could be added, such as columns fenced divs, empty fenced divs, etc.

**Comment by jjallaire** [comment]
Another thing to note is that Quarto also has snippets which can be activated via the "Insert Snippet" command:

<img width="605" alt="Screen Shot 2022-11-11 at 1 41 43 PM" src="https://user-images.githubusercontent.com/104391/201409312-012198da-06ee-4254-b328-5f599b2f2171.png">
